subprojects/gst-plugins-good/gst/rtp/gstrtph266depay.c

```diff
@@ -0,0 +1,674 @@
+    return;
+  }
+}
+
+static void
+gst_rtp_h266_finish_fragmentation_unit (GstRtpH266Depay * rtph266depay)
+{
+  guint outsize;
+  GstBuffer *outbuf;
+
+  outsize = gst_adapter_available (rtph266depay->adapter);
+  g_assert (outsize >= sizeof (sync_bytes));
+
+  outbuf = gst_adapter_take_buffer (rtph266depay->adapter, outsize);
+  GST_DEBUG_OBJECT (rtph266depay, "output %d bytes", outsize);
+
+#if 0
+  {
+    GstMapInfo map;
+
+    gst_buffer_map (outbuf, &map, GST_MAP_READ);
+    g_assert (g_memcmp (map.data, sync_bytes, sizeof (sync_bytes)) == 0);
+    gst_buffer_unmap (outbuf, &map);
+  }
+#endif
+
+  rtph266depay->current_fu_type = 0;
+
+  gst_rtp_h266_depay_handle_nal (rtph266depay, outbuf,
+      rtph266depay->fu_timestamp, rtph266depay->fu_marker);
+}
+
+static GstBuffer *
+gst_rtp_h266_depay_process (GstRTPBaseDepayload * depayload, GstRTPBuffer * rtp)
+{
+  GstRtpH266Depay *rtph266depay = GST_RTP_H266_DEPAY (depayload);
+  GstBuffer *outbuf = NULL;
+  GstMapInfo map;
+  gint payload_len;
+  guint8 *payload;
+  guint8 nal_unit_type, nuh_layer_id, nuh_temporal_id_plus1;
+  guint header_len;
+  GstClockTime timestamp;
+  gboolean marker;
+  guint outsize, nalu_size;
+
+  GST_DEBUG_OBJECT (depayload, "Start processing.");
+
+  payload_len = gst_rtp_buffer_get_payload_len (rtp);
+  payload = gst_rtp_buffer_get_payload (rtp);
+
+  GST_DEBUG_OBJECT (rtph266depay, "receiving %d bytes", payload_len);
+
+  if (payload_len == 0)
+    goto empty_packet;
+
+  // +---------------+---------------+
+  // |0|1|2|3|4|5|6|7|0|1|2|3|4|5|6|7|
+  // +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+  // |F|Z| LayerID   |  Type   | TID |
+  // +---------------+---------------+
+
+  /* *INDENT-OFF* */
+  nal_unit_type = (payload[1] >> 3) & 0b00011111;
+  nuh_layer_id = payload[0] & 0b00111111;
+  nuh_temporal_id_plus1 = payload[1] & 0b00000111;
+  /* *INDENT-ON* */
+
+  header_len = 2;
+
+  timestamp = GST_BUFFER_PTS (rtp->buffer);
+  marker = gst_rtp_buffer_get_marker (rtp);
+  GST_INFO_OBJECT (rtph266depay, "marker: %d", marker);
+
+  GST_INFO_OBJECT (rtph266depay,
+      "NAL header nal_unit_type %d, nuh_temporal_id_plus1 %d", nal_unit_type,
+      nuh_temporal_id_plus1);
+  GST_DEBUG_OBJECT (depayload, "is discont %d",
+      GST_BUFFER_IS_DISCONT (rtp->buffer));
+
+  GST_FIXME_OBJECT (rtph266depay, "Assuming DONL field is not present");
+
+
+  /* If FU unit was being processed, but the current nal is of a different
+   * type.  Assume that the remote payloader is buggy (didn't set the end bit
+   * when the FU ended) and send out what we gathered thusfar */
+  if (G_UNLIKELY (rtph266depay->current_fu_type != 0 &&
+          nal_unit_type != rtph266depay->current_fu_type)) {
+    gst_rtp_base_depayload_delayed (depayload);
+    gst_rtp_h266_finish_fragmentation_unit (rtph266depay);
+  }
+
+  switch (nal_unit_type) {
+    case GST_H266_NAL_AP:
+    {
+      // TODO: Payloader does not support Aggregation packets yet.
+      g_assert_not_reached ();
```

Comment:
It's better to log an error and return `NULL` here. This path is correct; it's just not implemented. We must abort the pipeline in a controlled way.

Note that you should think that every `g_assert_* ()` can be disabled (deleted from the code) with `G_DISABLE_ASSERT`.

Reply:
Also, I think this error path might deal with some resource liberation (like `rtp` or `payload`).

Reply:
I added not_implemented goto statement.